packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2277,7 +2277,7 @@ class IterableProperty<T> extends DiagnosticsProperty<Iterable<T>> {
 ///
 ///  * [DiagnosticsProperty] which documents named parameters common to all
 ///    [DiagnosticsProperty].
-class EnumProperty<T> extends DiagnosticsProperty<T> {
+class EnumProperty<T> extends DiagnosticsProperty<Enum> {
```

Comment:
This appears to be a breaking change

Reply:
Could also be `EnumProperty<Enum> extends DiagnosticsProperty<T>`, what do you think?

Reply:
I don't know if customer_testing is a flake or what is happening there (the logs show success only, as far as I found), but I sent a PR to plugins already. Plugins don't use EnumProperty.

Reply:
Anyone who is using EnumProperty<T> will be broken though. Should this be deprecated as well in favor of something new?

Reply:
I don't think so, because EnumProperty will be enforced to receive Enums, right now it is not enforced. It is used in tests 98% of the time, and by Flutter itself. I doubt it will break someone.

What would you do?

Reply:
> What would you do?

I would not assume.

Reply:
I just checked here and all other diagonistcs have a strong type:

<img width="531" alt="image" src="https://user-images.githubusercontent.com/351125/232902022-7f7a35ad-781b-4e37-b25a-bbb5bc919224.png">

Reply:
I searched here, most occurrences seems to be Flutter only, and these won't change anything because Flutter already uses Enum everywhere. https://github.com/search?type=code&auto_enroll=true&q=EnumProperty